x/concentrated-liquidity/incentives_test.go

```diff
@@ -2908,14 +2908,33 @@ func (s *KeeperTestSuite) TestCollectIncentives() {
 			}
 
 			// System under test
-			actualIncentivesClaimed, actualIncetivesForfeited, err := clKeeper.CollectIncentives(ctx, ownerWithValidPosition, DefaultPositionId)
-
-			// Assertions
+			incentivesClaimedQuery, incentivesForfeitedQuery, err := clKeeper.GetClaimableIncentives(ctx, DefaultPositionId)
 
 			poolBalanceAfterCollect := s.App.BankKeeper.GetAllBalances(ctx, validPool.GetAddress())
 			incentivesBalanceAfterCollect := s.App.BankKeeper.GetAllBalances(ctx, validPool.GetIncentivesAddress())
 			ownerBalancerAfterCollect := s.App.BankKeeper.GetAllBalances(ctx, ownerWithValidPosition)
 
+			if tc.expectedError != nil {
+				s.Require().Error(err)
+				s.Require().ErrorContains(err, tc.expectedError.Error())
+				s.Require().Equal(tc.expectedIncentivesClaimed, incentivesClaimedQuery)
+				s.Require().Equal(tc.expectedForfeitedIncentives, incentivesForfeitedQuery)
+
+				// Ensure balances are unchanged
+				s.Require().Equal(incentivesBalanceBeforeCollect, incentivesBalanceAfterCollect)
+				s.Require().Equal(ownerBalancerAfterCollect, ownerBalancerBeforeCollect)
```

Comment:
We should probably be checking this even if no error

Reply:
Good point, changed here https://github.com/osmosis-labs/osmosis/pull/5007/commits/6a164cee29e89bf304b495059a8eb4f60add6710